openedx/frontend-app-learning
# Endpoint usage implementation for Courseware Search

## Description
Ticket: [KBK-45 🔒](https://2u-internal.atlassian.net/browse/KBK-45)

This change takes the the [Search Results UI components 🔒](https://2u-internal.atlassian.net/wiki/spaces/Kubrick/pages/614432780/Component+breakdown+based+on+Figma) and connects it to the search endpoint. I'm taking some small liberties on the design and added some extra changes to the components due to some unplanned adaptations I had to make.

> [!IMPORTANT]  
> These changes add he `joi` dependency to be used on payload validation.

## UI changes

### Showing results
![courseware-search-screenshot](https://github.com/openedx/frontend-app-learning/assets/12736783/dd6ea911-8308-4e98-90fb-1c480f623161)

## Added error alert in case of endpoint failure or wrong payload format.
![search-error](https://github.com/openedx/frontend-app-learning/assets/12736783/144f1f20-a6df-4955-95cd-159ed8eaae4c)



## Testing Instructions:

This feature is behind a Waffle Flag. In order to test it:
- Follow the instructions to enable the waffle flags as indicated in https://github.com/openedx/frontend-app-learning/pull/1199.
- A new Search icon should show on the right side of the navigation menu on the Course Home page.
